x/profile/internal/types/msgs.go

```diff
@@ -0,0 +1,206 @@
+)
+
+// ----------------------
+// --- MsgCreateProfile
+// ----------------------
+
+// MsgCreateProfile defines a CreateProfile message
+type MsgCreateProfile struct {
+	Name     string         `json:"name,omitempty"`
+	Surname  string         `json:"surname,omitempty"`
+	Moniker  string         `json:"moniker"`
+	Bio      string         `json:"bio,omitempty"`
+	Pictures *Pictures      `json:"pictures,omitempty"`
+	Creator  sdk.AccAddress `json:"creator"`
+}
+
+// NewMsgCreateProfile is a constructor function for MsgCreateProfile
+func NewMsgCreateProfile(name string, surname string, moniker string, bio string, pictures *Pictures,
+	creator sdk.AccAddress) MsgCreateProfile {
+	return MsgCreateProfile{
+		Name:     name,
+		Surname:  surname,
+		Moniker:  moniker,
+		Bio:      bio,
+		Pictures: pictures,
+		Creator:  creator,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgCreateProfile) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgCreateProfile) Type() string { return ActionCreateProfile }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgCreateProfile) ValidateBasic() error {
+	if msg.Creator.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("Invalid creator address: %s", msg.Creator))
+	}
+
+	if len(msg.Name) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Surname) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(strings.TrimSpace(msg.Moniker)) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Profile moniker cannot be blank or empty")
+	}
+
+	if len(msg.Moniker) > MaxMonikerLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile moniker cannot exceed %d characters", MaxMonikerLength))
+	}
+
+	if len(msg.Bio) > MaxBioLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile biography cannot exceed %d characters", MaxBioLength))
+	}
+
+	return nil
+}
+
+// GetSignBytes encodes the message for signing
+func (msg MsgCreateProfile) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
+
+// GetSigners defines whose signature is required
+func (msg MsgCreateProfile) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Creator}
+}
+
+// ----------------------
+// --- MsgEditProfile
+// ----------------------
+
+// MsgEditProfile defines a EditProfile message
+type MsgEditProfile struct {
+	PreviousMoniker string         `json:"previous_moniker"`
+	NewMoniker      string         `json:"new_moniker"`
+	Name            string         `json:"name,omitempty"`
+	Surname         string         `json:"surname,omitempty"`
+	Bio             string         `json:"bio,omitempty"`
```

Comment:
I'm not sure that we want to use `string` here. Maybe `*string` might be better as it can represents better (using `nil`) the willing of the user to not edit the specific value. So that: 

- if the value is `nil`, do not edit it
- otherwise, use the one specified

Reply:
This looks still open

Reply:
@bragaz [Here](https://github.com/desmos-labs/desmos/pull/122#discussion_r403928905)'s the answer box. I'll copy your answer here: 

> As I said to you on last friday call, I will pass the default string value instead.
I've done this just to keep the cli commands clean otherwise I should have put a lot of if statements just to check if a field has the default and than put it to nil before build the message.

By leaving the `Msg` with `string`, it means that you are moving these checks into the `Handler` tho. What it could be done is creating a method that acts like that: 

```golang
func GetFlagValueOrNilOnDefault(flag string) *string {
  var flagValue string* := &viper.GetString(flag)
  if flagValue == "default" {
    flagValue = nil
  }
  return flagValue
}
```

Then you could call this easily by doing 

```golang
GetFlagValueOrNilOnDefault(flagName) 
GetFlagValueOrNilOnDefault(flagSurname) 
```